File: payroll_saas_backend/app/api/employees.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.models.schemas import EmployeeCreate, EmployeeUpdate
from app.services import employee_service
from app.core.database import SessionLocal

router = APIRouter(prefix="/employees", tags=["Employees"])

def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

@router.post("/")
def add_employee(payload: EmployeeCreate, db: Session = Depends(get_db)):
    return employee_service.create_employee(db, payload)

@router.get("/")
def get_employees(db: Session = Depends(get_db)):
    return employee_service.list_employees(db)

@router.get("/{employee_id}")
def get_employee(employee_id: int, db: Session = Depends(get_db)):
    return employee_service.get_employee(db, employee_id)

@router.put("/{employee_id}")
def update_employee(employee_id: int, payload: EmployeeUpdate, db: Session = Depends(get_db)):
    return employee_service.update_employee(db, employee_id, payload)

@router.delete("/{employee_id}")
def delete_employee(employee_id: int, db: Session = Depends(get_db)):
    employee_service.delete_employee(db, employee_id)
    return {"message": "Employee deleted successfully"}

File: payroll-saas-frontend/src/api/employees.ts
import axios from "axios";

const API_URL = "http://localhost:8000/employees/";

export const getEmployees = async () => {
  const res = await axios.get(`${API_URL}`);
  return res.data;
};

export const createEmployee = async (data: any) => {
  const res = await axios.post(`${API_URL}`, data);
  return res.data;
};

export const updateEmployee = async (id: number, data: any) => {
  const res = await axios.put(`${API_URL}${id}`, data);
  return res.data;
};

export const deleteEmployee = async (id: number) => {
  const res = await axios.delete(`${API_URL}${id}`);
  return res.data;
};